kolibri/core/logger/api.py

```diff
@@ -913,4 +974,50 @@ class AttemptLogViewSet(ReadOnlyValuesViewset):
         "error",
         "masterylog",
         "sessionlog",
+        "diff__correct",
     )
+    field_map = {"diff": _attempts_diff}
+
+    def annotate_queryset(self, queryset):
+        if "diff__correct" not in queryset.query.annotations:
+            queryset = queryset.annotate(
+                diff__correct=Value(None, output_field=IntegerField())
+            )
+        return queryset
+
+    @action(detail=False)
+    def diff(self, request):
+        # protect against expensive queries from loose unsupported filtering
+        min_required_filter_sets = (
+            {"masterylog"},
+            {"content", "item"},
+            {"content", "user"},
+        )
+        for filter_set in min_required_filter_sets:
```

Comment:
Nice - the required parameters stuff I'd done previously that I linked above didn't allow for multiple disjoint sets - and then the required parameters I did in [Studio](https://github.com/learningequality/studio/blob/unstable/contentcuration/contentcuration/viewsets/base.py#L838) was just preventing no queries.

The other alternative might be to enforce cursor pagination on endpoints.